frame/elections-phragmen/src/benchmarking.rs

```diff
@@ -0,0 +1,283 @@
+	let entropy = (name, index).using_encoded(blake2_256);
+	T::AccountId::decode(&mut &entropy[..]).unwrap_or_default()
+}
+
+/// grab new account with infinite balance.
+fn endowed_account<T: Trait>(name: &'static str, index: u32) -> T::AccountId {
+	let account = account::<T>(name, index);
+	let _ = T::Currency::make_free_balance_be(&account, BalanceOf::<T>::max_value());
+	account
+}
+
+/// Account ot lookup type of system trait.
+fn as_lookup<T: Trait>(account: T::AccountId) -> Lookup<T> {
+	T::Lookup::unlookup(account)
+}
+
+/// Get a reasonable amount of stake based on the execution trait's configuration
+fn default_stake<T: Trait>(factor: u32) -> BalanceOf<T> {
+	let factor = BalanceOf::<T>::from(factor);
+	T::Currency::minimum_balance() * factor
+}
+
+/// Add `c` new candidates.
+fn submit_candidates<T: Trait>(c: u32) -> Result<Vec<T::AccountId>, &'static str> {
+	(0..c).map(|i| {
+		let account = endowed_account::<T>("candidate", i);
+		<Elections<T>>::submit_candidacy(RawOrigin::Signed(account.clone()).into())
+			.map_err(|_| "failed to submit candidacy")?;
+		Ok(account)
+	}).collect::<Result<_, _>>()
+}
+
+/// Submit one voter.
+fn submit_voter<T: Trait>(caller: T::AccountId, votes: Vec<T::AccountId>, stake: BalanceOf<T>)
+	-> Result<(), &'static str>
+{
+	<Elections<T>>::vote(RawOrigin::Signed(caller).into(), votes, stake)
+		.map_err(|e| "failed to submit vote")
+}
+
+benchmarks! {
+	_ {}
+
+	// -- Signed ones
+	vote {
+		// range of candidates to vote for. Direct argument of the dispatchable.
+		let x in 1 .. (MAXIMUM_VOTE as u32);
+
+		// create a bunch of candidates.
+		let all_candidates = submit_candidates::<T>(MAXIMUM_VOTE as u32)?;
+
+		let caller = endowed_account::<T>("caller", SEED);
+		let stake = default_stake::<T>(BALANCE_FACTOR);
+
+		// vote first `x` ones.
+		let votes = all_candidates.into_iter().take(x as usize).collect();
+	}: _(RawOrigin::Signed(caller), votes, stake)
+
+	remove_voter {
+		// No complexity parameter. while we can vote for numerous candidates and then remove them
+		// (and I think each can have slightly different performance), we cannot express this via
+		// the parameters. Hence, we only check for the median, 8 votes to be removed. Only use
+		// account seed as complexity parameter.
+		// TODO: what should be the upper bound here? the first benchmark runs 16 iterations.. this
+		// one 1000? These are used only to iterate and run the benchmark s times. They are not a
+		// parameter in any way.
+		let s in 0 .. 1000;
+
+		let votes_to_remove = (MAXIMUM_VOTE / 2) as u32;
+
+		// create a bunch of candidates.
+		let all_candidates = submit_candidates::<T>(votes_to_remove)?;
+
+		let caller = endowed_account::<T>("caller", s);
+		let stake = default_stake::<T>(BALANCE_FACTOR);
+		submit_voter::<T>(caller.clone(), all_candidates, stake)?;
+	}: _(RawOrigin::Signed(caller))
+
+	report_defunct_voter {
+		// has no complexity parameter. two voters exist. One only votes fro outdated candidates.
```

Comment:
This is not accurate. You write in your code

```
			// Checking if someone is a candidate and a member here is O(LogN), making the whole
			// function O(MLonN) with N candidates in total and M of them being voted by `target`.
```

and `is_defunct_voter` iterates over `votes_of(who)`, so you should populate this vec using component paramters.

Reply:
is defunct voter also does a binary search over the candidates and members and runner. So you need to populate all of that